packages/pipeline/src/entities/token_order.ts

```diff
@@ -0,0 +1,27 @@
+import { BigNumber } from '@0x/utils';
+import { Column, Entity, PrimaryColumn } from 'typeorm';
+
+import { OrderType } from '../types';
+import { bigNumberTransformer } from '../utils';
+
+@Entity({ name: 'token_orderbook_snapshots', schema: 'raw' })
+export class TokenOrderbookSnapshot {
+    @PrimaryColumn({ name: 'source' })
+    public source!: string;
+    @PrimaryColumn({ name: 'retrieval_timestamp', type: 'numeric', transformer: bigNumberTransformer })
```

Comment:
`retrieval_timestamp` doesn't need to have type `BigNumber`/`numeric`. You can use `number` for the JavaScript type and `bigint` for the Postgres type like we do for all other timestamps. The `number` type in JavaScript gives us guaranteed precision for integers up to a maximum up to 2^53-1. This is more than enough to cover Unix timestamps in milliseconds for the next few centuries 😄

Reply:
But we're going to need this to run for thousands of millennia!!